In [1]:
import pandas as pd

# Load dataset
df = pd.read_csv("books_data.csv")

# First 5 records
print("First 5 Records:")
print(df.head())

# Dataset Information
print("\nDataset Information:")
print(df.info())

# Check Missing Values
print("\nMissing Values:")
print(df.isnull().sum())

# Basic Statistics
print("\nPrice Statistics:")
print(df["Price"].describe())

# Rating Count
print("\nRating Distribution:")
print(df["Rating"].value_counts())

# Highest Price Book
print("\nMost Expensive Book:")
print(df.loc[df["Price"].idxmax()])

# Lowest Price Book
print("\nLeast Expensive Book:")
print(df.loc[df["Price"].idxmin()])

First 5 Records:
                                   Title  Price Rating
0                   A Light in the Attic  51.77  Three
1                     Tipping the Velvet  53.74    One
2                             Soumission  50.10    One
3                          Sharp Objects  47.82   Four
4  Sapiens: A Brief History of Humankind  54.23   Five

Dataset Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Title   20 non-null     object 
 1   Price   20 non-null     float64
 2   Rating  20 non-null     object 
dtypes: float64(1), object(2)
memory usage: 612.0+ bytes
None

Missing Values:
Title     0
Price     0
Rating    0
dtype: int64

Price Statistics:
count    20.000000
mean     38.048500
std      15.135231
min      13.990000
25%      22.637500
50%      41.380000
75%      51.865000
max      57.250000
Name: Price, dtype: float64

Rating Distribut

# Exploratory Data Analysis on Book Dataset

## Objective

To explore and analyze the scraped book dataset and identify useful insights.

## Analysis Performed

* Examined the structure of the dataset.
* Checked for missing values.
* Calculated summary statistics for book prices.
* Analyzed the distribution of ratings.
* Identified the most expensive and least expensive books.

## Key Findings
Key Findings

• The dataset contains 20 books with title, price, and rating information.
• No missing values were found in the dataset.
• The average book price is 38.05.
• The most expensive book costs 57.25, while the least expensive book costs 13.99.
• One-star rated books are the most common category in the dataset.
• The dataset is clean and suitable for visualization and further analysis.